### PR TITLE
Added maxdf option to fusedlasso and dualpathFused functions; this st…

### PR DESCRIPTION
**Pull request by ajgreen93**
…ops the path algorithm once the computed solution has at least maxdf + 1 degrees of freedom.

### CODE REVIEW

File: R/dualpathFused.R [hunk 1]
```diff
@@ -339,6 +339,15 @@ dualpathFused <- function(y, D, approx=FALSE, maxsteps=2000, minlam=0,
     }
     completepath = FALSE
   }
+  
+  # If we reached the maximum degrees of freedom
+  else if (df[k-1]>maxdf) {
+    if (verbose) {
+      cat(sprintf("\nReached the maximum degrees of freedom (%.3f),",maxdf))
```
Comment (ryantibs) [inline R/dualpathFused.R]: You should change `%.3f` here to `%i` (and test it out). The former is for floats (and specifies that we should show 3 decimals); the latter is for integers.

File: R/dualpathFused.R [hunk 2]
```diff
@@ -10,7 +10,7 @@
 # the open interval to the *right* of the current lambda_k.
 
 dualpathFused <- function(y, D, approx=FALSE, maxsteps=2000, minlam=0,
-                          rtol=1e-7, btol=1e-7, verbose=FALSE,
+                          maxdf=length(y), rtol=1e-7, btol=1e-7, verbose=FALSE,
```
Comment (ryantibs) [inline R/dualpathFused.R:13]: I can't tell by eye but please make sure the column width doesn't go over 80 characters (that's the coding style with the rest of the package), and if so, just put the `verbose` argument on the next line.

File: R/dualpathFused.R (hunk 1 above)
Comment (ajgreen93) [reply R/dualpathFused.R]: Ugh, sloppy. Fixed.

File: R/dualpathFused.R (hunk 2 above)
Comment (ajgreen93) [reply R/dualpathFused.R:13]: 79 :)

**Pull request merged**